src/Form/matching.js

```diff
@@ -968,8 +970,15 @@ const getLargestMeaningfulContainer = (el, form, cssSelector) => {
     const inputsInParentsScope = parentElement.querySelectorAll(cssSelector)
     // To avoid noise, ensure that our input is the only in scope
     if (inputsInParentsScope.length === 1) {
+        // If the parent has only 1 input and a label with text, we've found our meaningful container
+        const labelInParentScope = parentElement.querySelector('label')
```

Comment:
Looks like this would find a `<label/>` at any depth within `parentElement`, something like:

```html
<div id="parentElement">
  <span>
    <label/>
  </span>
  <input>
</div>
```

I assume that's intentional 🙂

Reply:
Yes, that's intentional 🙂.